packages/gatsby-plugin-sitemap/src/__tests__/gatsby-node.js

```diff
@@ -176,5 +176,36 @@ describe(`Test plugin sitemap`, async () => {
       expect(filePath).toEqual(path.join(`public`, `sitemap.xml`))
       expect(contents).toMatchSnapshot()
     })
+    it(`should include path prefix when creating creating index sitemap`, async () => {
+      const mockStream = {
+        once: jest.fn((event, cb) => cb()),
+        end: jest.fn(),
+        write: jest.fn(),
+      }
+
+      const sitemapIndexPath = `./public/sitemap-index.xml`
+      fs.createWriteStream.mockReturnValue(mockStream)
+      const options = {
+        sitemapSize: 1,
+      }
+      const prefix = `/test`
+      await onPostBuild({ graphql, pathPrefix: prefix }, options)
+
+      expect(fs.createWriteStream.mock.calls[2][0]).toEqual(sitemapIndexPath)
+
+      const parser = new DOMParser()
+      const DOM = parser.parseFromString(
+        mockStream.write.mock.calls[2][0],
+        `application/xml`
+      )
+      const elements = DOM.querySelectorAll(`sitemap loc`)
+      expect(elements.length).toBe(2)
+      expect(elements[0].innerHTML).toBe(
+        queryResult.data.site.siteMetadata.siteUrl + prefix + `/sitemap-0.xml`
+      )
+      expect(elements[1].innerHTML).toBe(
+        queryResult.data.site.siteMetadata.siteUrl + prefix + `/sitemap-1.xml`
+      )
```

Comment:
perhaps we should just test if `createSitemapIndex` is called with `hostname` that includes pathPrefix? This mocks some internals of `sitemap` package that can change, making this test not that maintainable

Reply:
Thanks @pieh, I've updated the test to reflect your feedback.